pkg/command/root.go

```diff
@@ -253,12 +269,42 @@ func duplicateAliasWarning(rootCmd *cobra.Command) {
 		}
 	}
 }
+func ensureCLIInstanceID() (string, error) {
+	cliID, _ := config.GetCLIId()
+	if cliID != "" {
+		return cliID, nil
+	}
+	cliID = uuid.New().String()
+	err := config.SetCLIId(cliID)
+	if err != nil {
+		return "", err
+	}
+	return cliID, nil
+}
 
 // Execute executes the CLI.
 func Execute() error {
 	root, err := NewRootCmd()
 	if err != nil {
 		return err
 	}
-	return root.Execute()
+	err = root.Execute()
+	exitCode := 0
+	if err != nil {
+		exitCode = 1
+		if errStr, ok := err.(*exec.ExitError); ok {
+			// If a plugin exited with an error, we don't want to print its
+			// exit status as a string, but want to use it as our own exit code.
+			exitCode = (errStr.ExitCode())
+		}
+	}
+
+	postRunMetrics := &telemetry.PostRunMetrics{ExitCode: exitCode}
+	if err := telemetry.Client().UpdateCmdPostRunMetrics(postRunMetrics); err != nil {
```

Comment:
Nit:IMO, I would suggest renaming the error variables here for better readability. As we are returning the `err = root.Execute()` at the end.